src/smhi/smhi.py

```diff
@@ -76,116 +66,145 @@ def find_stations_from_gps(
+        if distance > 0:
             # Find the station latitude and longitude information from Metobs
-            stat = next(
-                item for item in self.client.stations.stations if item["id"] == station
+            # should be replaced by a self.periods.position[0].latitude
+            latitude, longitude = (
+                periods.position[0].latitude,
+                periods.position[0].longitude,
             )
-            latitude = stat["latitude"]
-            longitude = stat["longitude"]
 
-            holes_to_fill = data[
-                data.index.to_series().diff() > data.index.to_series().diff().median()
+            holes_to_fill = data.df[
+                data.df.index.to_series().diff()
+                > data.df.index.to_series().diff().median()
             ]
-            # Find stations within a given radius - set in "interpolate".
-            self.find_stations_from_gps(
-                parameter=parameter,
-                latitude=latitude,
-                longitude=longitude,
-                dist=interpolate,
+
+            # Find stations within a given radius - set in "distance".
+            nearby_stations = self._find_stations_from_gps(
+                stations, latitude, longitude, distance
             )
 
             # Iterate over nearby stations, starting with the closest
-            for nearby_station in self.nearby_stations[1:]:
-                tmpdata, _ = self.get_data(parameter, nearby_station[0])
-                for time, _ in holes_to_fill.iterrows():
-                    earliertime = data[data.index < time].index.max()
+            for nearby_station in nearby_stations[1:]:
+                nearby_data = Data(Periods(stations, nearby_station[0]))
 
-                    if (
-                        len(
-                            tmpdata[
-                                (tmpdata.index > earliertime) & (tmpdata.index < time)
-                            ]
+                for time, _ in holes_to_fill.iterrows():
+                    earliertime = data.df[data.df.index < time].index.max()
+                    condition = (nearby_data.df.index > earliertime) & (
+                        nearby_data.df.index < time
+                    )
+
+                    if len(nearby_data.df[condition]) > 0:
+                        data.df = pd.concat(
+                            [data.df, nearby_data.df], axis=0, join="outer"
                         )
-                        > 0
-                    ):
-                        data = pd.concat([data, tmpdata], axis=0, join="outer")
 
                 # Re-check how many holes remain
-                holes_to_fill = data[
-                    data.index.to_series().diff()
-                    > data.index.to_series().diff().median()
+                holes_to_fill = data.df[
+                    data.df.index.to_series().diff()
+                    > data.df.index.to_series().diff().median()
```

Comment:
Could you name this as a variable, so others can easer understand what it compares?